http/oidc/src/test/java/org/wildfly/security/http/oidc/AbstractLogoutTest.java

```diff
@@ -0,0 +1,240 @@
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.wildfly.security.http.HttpAuthenticationException;
+import org.wildfly.security.http.HttpConstants;
+import org.wildfly.security.http.HttpScope;
+import org.wildfly.security.http.HttpServerAuthenticationMechanism;
+import org.wildfly.security.http.Scope;
+
+/**
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public abstract class AbstractLogoutTest extends OidcBaseTest {
+
+    private ElytronDispatcher dispatcher;
+    private OidcClientConfiguration clientConfig;
+
+    @BeforeClass
+    public static void onBeforeClass() {
+        assumeTrue("Docker isn't available, OIDC tests will be skipped", isDockerAvailable());
+        KEYCLOAK_CONTAINER = new KeycloakContainer();
+        KEYCLOAK_CONTAINER.start();
+        System.setProperty("oidc.provider.url", KEYCLOAK_CONTAINER.getAuthServerUrl() + "/realms/" + TEST_REALM);
+    }
+
+    @AfterClass
+    public static void onAfterClass() {
+        System.clearProperty("oidc.provider.url");
+    }
+
+    @AfterClass
+    public static void generalCleanup() {
+        // no-op
+    }
+
+    @Before
+    public void onBefore() throws Exception {
+        OidcBaseTest.client = new MockWebServer();
+        OidcBaseTest.client.start(new InetSocketAddress(0).getAddress(), CLIENT_PORT);
+        configureDispatcher();
+        RealmRepresentation realm = KeycloakConfiguration.getRealmRepresentation(
+                TEST_REALM, CLIENT_ID, CLIENT_SECRET, CLIENT_HOST_NAME, CLIENT_PORT,
+                CLIENT_APP, false);
+
+        // values set high to ensure a slow test env will succeed.
+        realm.setAccessTokenLifespan(100);
+        realm.setSsoSessionMaxLifespan(100);
+
+        ClientRepresentation client = realm.getClients().get(0);
+
+        client.setAttributes(new HashMap<>());
+
+        doConfigureClient(client);
+
+        List<String> redirectUris = new ArrayList<>(client.getRedirectUris());
+
+        redirectUris.add("*");
+
+        client.setRedirectUris(redirectUris);
+
+        sendRealmCreationRequest(realm);
+    }
+
+    @After
+    public void onAfter() throws IOException {
+        client.shutdown();
+        RestAssured
+                .given()
+                .auth().oauth2(KeycloakConfiguration.getAdminAccessToken(KEYCLOAK_CONTAINER.getAuthServerUrl()))
+                .when()
+                .delete(KEYCLOAK_CONTAINER.getAuthServerUrl() + "/admin/realms/" + TEST_REALM).then().statusCode(204);
+    }
+
+    protected void doConfigureClient(ClientRepresentation client) {
+    }
+
+    protected OidcJsonConfiguration getClientConfiguration() {
+        OidcJsonConfiguration config = new OidcJsonConfiguration();
+
+        config.setRealm(TEST_REALM);
+        config.setResource(CLIENT_ID);
+        config.setPublicClient(false);
+        config.setAuthServerUrl(KEYCLOAK_CONTAINER.getAuthServerUrl());
```

Comment:
Would be good to use setProviderUrl and setClientId instead of setRealm, setResource, and setAuthServerUrl

Reply:
changed